onnxruntime/core/framework/execution_provider.cc

```diff
@@ -231,18 +246,28 @@ int IExecutionProvider::ModelMetadefIdGenerator::GenerateId(const onnxruntime::G
   }
 
   // return the current unique id, and increment to update
-  return model_metadef_id_[model_hash]++;
+  return model_id_[model_hash]++;
 }
 
 int IExecutionProvider::GenerateMetaDefId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash) const {
-  ORT_ENFORCE(metadef_id_generator_,
-              "IExecutionProvider constructor must be called with true for use_metadef_id_creator");
+  ORT_ENFORCE(model_id_generator_,
+              "IExecutionProvider constructor must be called with true for use_model_id_creator");
 
   // if the EP is shared across multiple sessions there's a very small potential for concurrency issues.
   // use a lock when generating an id to be paranoid
   static OrtMutex mutex;
   std::lock_guard<OrtMutex> lock(mutex);
-  return metadef_id_generator_->GenerateId(graph_viewer, model_hash);
+  return model_id_generator_->GenerateId(graph_viewer, model_hash);
 }
 
+int IExecutionProvider::GenerateModelId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash) const {
+  ORT_ENFORCE(model_id_generator_,
+              "IExecutionProvider constructor must be called with true for use_model_id_creator");
+
+  // if the EP is shared across multiple sessions there's a very small potential for concurrency issues.
```

Comment:
The EP is owned by a single session. What's the use case for it being shared between multiple sessions?